File: app/flytec/lib/features/aplications_v2/models/aplicacao_relatorio.dart
import 'package:flytec/features/aplications_v2/models/aplicacao.dart';

class AplicacaoRelatorio {
  String cultura;
  String produtoAplicado;
  String dosagem;
  String unidadeDosagem;
  String volumeAplicacao;
  String unidadeVolumeAplicacao;
  String totalAreaAplicadaHa;
  String localizacaoPistaCodigoICAO;
  String latitudeSul;
  String longitudeOeste;
  String densidade;
  String alteracoesPlanejamentoObservacoes;
  String relatorioDGPSLogs;
  List<Aplicacao> aplicacoes;

  AplicacaoRelatorio({
    required this.cultura,
    required this.produtoAplicado,
    required this.dosagem,
    required this.unidadeDosagem,
    required this.volumeAplicacao,
    required this.unidadeVolumeAplicacao,
    required this.totalAreaAplicadaHa,
    required this.localizacaoPistaCodigoICAO,
    required this.latitudeSul,
    required this.longitudeOeste,
    required this.densidade,
    required this.alteracoesPlanejamentoObservacoes,
    required this.relatorioDGPSLogs,
    required this.aplicacoes,
  });
}

File: backoffice/Domain/Entidades/Cadastros/Aplicacao/AplicacaoRelatorio.cs
using Domain.Entidades.Cadastros.Pistas;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Domain.Entidades.Cadastros.Aplicacao;

public class AplicacaoRelatorio
{
    [Key]
    public int Id { get; set; }

    [ForeignKey("Aplicacao")]
    public int? IdAplicacao { get; set; }

    [ForeignKey("Pista")]
    public int? IdPista { get; set; }
    public decimal? Dosagem { get; set; }
    public string KG_LT { get; set; }
    public int? VolumeAplicacao { get; set; }
    public decimal? TotalAreaAplicada { get; set; }
    public string? Alteracoes_Observacoes { get; set; }
    public string? Cultura { get; set; }
    public string? ProdutoAplicado { get; set; }
    public string? LocalizacaoPistaCodigoICAO { get; set; }
    public string? Latitude { get; set; }
    public string? Longitude { get; set; }
    public string? Densidade { get; set; }
    public string? RelatorioDGPS { get; set; }
    public string? UnidadeVolumeAplicacao { get; set; }
    public string? MapaAplicacao { get; set; }

    [ForeignKey("Empresa")]
    public int? IdEmpresa { get; set; }

    [JsonIgnore]
    public virtual Empresa.Empresa? Empresa { get; set; }

    [JsonIgnore]
    public virtual Aplicacao? Aplicacao { get; set; }
    [JsonIgnore]
    public virtual Pista? Pista { get; set; }
}
